spec/parser/options_spec.rb

```diff
@@ -425,6 +428,7 @@ def remaining
 
       it "accepts a switch=<value> assignment" do
         expect(parse("--attributes=a", "b", "c")["attributes"]).to eq(%w(a b c))
+        expect(parse("--attributes=-a", "b", "-c")["attributes"]).to eq(%w(-a b))
```

Comment:
This seems confusing... if "-a" and "-c" are valid values for an array attribute, then why would both "-a b" and "-a b -c" give you two values?

Reply:
The alternative is more confusing IMO and breaks backwards compatibility, i.e. `--attributes=a b -c` parsing to `{"attributes" => ['a', 'b', '-c']}` means that you can no longer have any options at all after an array option, which is not what the user expects.

This is the same reason this PR is limiting its scope to require a `=` to trigger this "leading `-` is a valid value" behavior.

Reply:
OK - I think we'd need to document this specifically then.

There seems to be another PR to fix the same issue? https://github.com/erikhuda/thor/pull/498 What's the difference?

Reply:
Yes, this aims to fix the same issue as #498, but #498 appears to have been abandoned. I believe it also has a bug as described by [this comment by @rafaelfranca](https://github.com/erikhuda/thor/pull/498#discussion_r100716476):

> I think setting this state here will broke it if there are more options after the first with - in the value, can you add tests to make sure it will work?

Reply:
Can you clarify what you're referring to by "option/options portion"? Is this something I should be adding as part of this PR?

Reply:
My bad - I forgot what project I was talking about 😛  The update would be to the Wiki once this is merged.